src/Nethermind/Nethermind.Mev.Test/BundlePoolTests.cs

```diff
@@ -19,87 +19,333 @@
+            testContext.BlockTree.NewHeadBlock += Raise.EventWith(new BlockEventArgs(Build.A.Block.WithNumber(test.Block).TestObject));
+            
+            List<MevBundle> bundlesForPastBlock = testContext.BundlePool.GetBundles(test.Block, test.TestTimestamp).ToList();
+            bundlesForPastBlock.Count.Should().Be(0);
+
+        }
+
+        [Test]
+        public void should_add_bundle_with_correct_timestamps()
+        {
+            ITimestamper timestamper = new ManualTimestamper(DateTime.UtcNow);
+            ulong timestamp = timestamper.UnixTime.Seconds;
+        
+            TestContext test = new TestContext(timestamper);
+
+            Transaction[] tx1 = new Transaction[]
+            {
+                Build.A.Transaction.SignedAndResolved(TestItem.PrivateKeyD).TestObject
+            };
+            Transaction[] tx2 = new Transaction[]
+            {
+                Build.A.Transaction.SignedAndResolved(TestItem.PrivateKeyA).TestObject
+            };
+            
+            MevBundle[] bundles = new []
+            {
+                new MevBundle(1, tx1, 0, 0), //should get added
+                new MevBundle(2, tx1, timestamp + 5, timestamp), //should not get added, min > max
+                new MevBundle(3, tx1, timestamp - 5,  timestamp + 5), //should get added
+                new MevBundle(4, tx1, timestamp + 4000, timestamp + 5000), //should not get added, min time too large
+                new MevBundle(4, tx2, timestamp, timestamp + 10), //should get added
+                new MevBundle(5, tx2, timestamp + 1, timestamp + 10) //should not get added, min timestamp too large
+                
+            };
+
+            bundles.Select(b => test.BundlePool.AddBundle(b))
+                .Should().BeEquivalentTo(new bool[] {true, false, true, false, true, false});
+        }
+
+        [Test]
+        public void should_reject_bundle_without_transactions()
+        {
+           TestContext tc = new();
+           Transaction[] emptyArr = Array.Empty<Transaction>();
+           Transaction[] filledArr = new Transaction[]
+           {
+               Build.A.Transaction.SignedAndResolved(TestItem.PrivateKeyD).TestObject,
+               Build.A.Transaction.SignedAndResolved(TestItem.PrivateKeyA).TestObject
+           };
+           
+           //Adding empty transaction array bundles
+           MevBundle bundleNoTx1 = new MevBundle(10, emptyArr);
+           MevBundle bundleNoTx2 = new MevBundle(11, emptyArr);
+           tc.BundlePool.AddBundle(bundleNoTx1).Should().BeFalse();
+           tc.BundlePool.AddBundle(bundleNoTx2).Should().BeFalse();
+           
+           //Same bundles with filled transaction array
+           MevBundle updatedBundle1 = new MevBundle(10, filledArr);
+           MevBundle updatedBundle2 = new MevBundle(11, filledArr);
+           tc.BundlePool.AddBundle(updatedBundle1).Should().BeTrue();
+           tc.BundlePool.AddBundle(updatedBundle2).Should().BeTrue();
```

Comment:
too many checks, checking different stuff, this should be 2 tests, with one check each